src/main/java/com/amazonaws/encryptionsdk/CreateDecryptingInputStreamRequest.java

```diff
@@ -20,12 +20,22 @@
 public class CreateDecryptingInputStreamRequest extends AwsCryptoRequest {
 
     private final InputStream inputStream;
+    private final Integer maxBodySize;
+    private final Integer maxHeaderSize;
 
     private CreateDecryptingInputStreamRequest(Builder builder) {
         super(builder);
 
         requireNonNull(builder.inputStream, "inputStream is required");
+        if (builder.maxBodySize != null && builder.maxBodySize < 0) {
+            throw new IllegalArgumentException("maxBodySize must be null or non-negative.");
+        }
+        if (builder.maxHeaderSize != null && builder.maxHeaderSize < 0) {
+            throw new IllegalArgumentException("maxHeaderSize must be null or positive.");
```

Comment:
Above you wrote "maxBodySize must be null or non-negative.".

Should this also read "maxHeaderSize must be null or non-negative"? As written, 0 is allowed here.